core/src/main/java/org/infinispan/functional/impl/MetaParamsExternalizer.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.functional.impl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.api.functional.MetaParam;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+public final class MetaParamsExternalizer extends AbstractExternalizer<MetaParams> {
```

Comment:
The problem with `Iterable` as return type is that the code calling `unmarshall` does not know the number of elements returned (and it's likely to have a use for that), while the `MarshallUtil` knows that. And actually the problem occurs with `marshall` as well, unless you provide number of elements ahead.

Any way, you're right that we should get rid of the externalizer and just expose static methods used by `MetaParamsInternalMetadata.Externalizer`